api/services/timelog.py

```diff
@@ -137,3 +139,73 @@ def check_task_for_overlap(self, task: Task) -> ValidatedObject:
                     f" {user_task_for_day.start_time} to {user_task_for_day.end_time}."
                 )
         return validated_task
+
+    def get_tasks_sum(self, user_id: int, start: date, end: date) -> int:
+        task_sum = (
+            self.db.query(func.sum(Task.task_total_minutes).label("task_sum"))
+            .filter(Task.user_id == user_id, Task.date.between(start, end))
+            .first()[0]
+        )
+        if task_sum is None:
+            task_sum = 0
+        return task_sum
+
+    def get_task_totals_projects(self, user_id: int, current_date: date) -> List[ProjectTaskSummary]:
+        totals = self.db.query(
+            func.public.get_user_project_summaries(user_id, current_date).table_valued(
+                "project_id", "project", "today_total", "today_text", "week_total", "week_text", "is_vacation"
+            )
+        ).all()
+        project_totals_list = []
+        for total in totals:
+            t = ProjectTaskSummary(
+                project_id=total[0],
+                project=total[1],
+                today_total=total[2],
+                today_text=total[3],
+                week_total=total[4],
+                week_text=total[5],
+                is_vacation=total[6],
+            )
+            if t.today_total is None:
+                t.today_total = 0
+            if t.week_total is None:
+                t.week_total = 0
+            project_totals_list.append(t)
+        return project_totals_list
+
+    def get_vacation_used(self, user_id: int, ref_date: date) -> int:
+        config = self.db.query(Config).first()
+        year_start = ref_date.replace(month=1, day=1)
+        used = (
+            self.db.query(func.sum(Task.task_total_minutes).label("vacation_used"))
+            .filter(Task.project_id == config.vacation_project_id, Task.date.between(year_start, ref_date))
+            .first()[0]
+        )
+        if used is None:
+            used = 0
+        return used
+
+    def get_vacation_scheduled(self, user_id: int, ref_date: date) -> int:
+        config = self.db.query(Config).first()
+        tomorrow = ref_date + timedelta(days=1)
+        yearEnd = ref_date.replace(month=12, day=31)
+        scheduled = (
+            self.db.query(func.sum(Task.task_total_minutes).label("vacation_sum"))
+            .filter(Task.project_id == config.vacation_project_id, Task.date.between(tomorrow, yearEnd))
+            .first()[0]
+        )
+        if scheduled is None:
+            scheduled = 0
+        return scheduled
+
+    def get_expected_worked_hours(self, user_capacities: List[UserCapacity], start_date: date, end_date: date) -> float:
```

Comment:
This is a function that doesn't need anything from the parent Class. What do you think of moving it to the helpers so we can also use it in other places like the project planning feature?

Reply:
That's a great idea!